CanvasPlusPlayground/Features/Announcements/CourseAnnouncementsView.swift

```diff
@@ -52,3 +41,82 @@ struct CourseAnnouncementsView: View {
     }
 }
 
+private struct AnnouncementRow: View {
+    let course: Course
+    let announcement: Announcement
+
+    var body: some View {
+        VStack(alignment: .announcementRowAlignment) {
+            header
+            detail
+        }
+    }
+
+    private var header: some View {
+        HStack {
+            Group {
+                if !(announcement.isRead ?? false) {
+                    Circle()
+                        .fill(.tint)
+                } else {
+                    Spacer().frame(width: unreadIndicatorWidth)
+                }
+            }
+            .frame(width: unreadIndicatorWidth, height: unreadIndicatorWidth)
+
+            Text(announcement.title ?? "")
+                .alignmentGuide(.announcementRowAlignment) { context in
+                    context[.leading]
+                }
+
+            Spacer()
+
+            if let createdAt = announcement.createdAt {
+                Text(createdAt.formatted(.relative(presentation: .named)))
+                    .foregroundStyle(.secondary)
+            }
+        }
+    }
+
+    private var detail: some View {
+        HStack {
+            Spacer().frame(width: unreadIndicatorWidth)
+
+            Group {
+                if let summary = announcement.summary {
+                    Text(Image(systemName: "wand.and.sparkles"))
+                        .foregroundStyle(
+                            course.rgbColors?.color ?? .accentColor
+                        )
+                    +
+                    Text(summary)
+                } else {
+                    AsyncAttributedText(
+                        announcement: announcement,
+                        textOnly: true
+                    )
+                }
+            }
+            .lineLimit(2)
+            .foregroundStyle(.secondary)
+            .controlSize(.small)
+            .alignmentGuide(.announcementRowAlignment) { context in
+                context[.leading]
+            }
+        }
+    }
+
+    private let unreadIndicatorWidth: CGFloat = 10
```

Comment:
Can properties like this be at the top

Reply:
Ah, sure. I like putting these "drawing constants" at the bottom lol